Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -0,0 +1,404 @@
+        /// <param name="syntaxConf">Object that contains the symbols and tokens for structuring the file syntax. The syntax configuration is represented by a <see cref="PxFileSyntaxConf"/> object.</param>
+        /// <param name="optionalEndSymbol">Optional symbol that closes the enclosement. If none given, startSymbol is used for both starting and ending the enclosement</param>
+        /// <return>Returns an <see cref="ExtractSectionResult"/> object that contains the extracted sections and the string that remains after the operation</return>
+        public static ExtractSectionResult ExtractSectionFromString(string input, char startSymbol, PxFileSyntaxConf syntaxConf, char? optionalEndSymbol = null)
+        {
+            // If no end symbol is provided, the start symbol is used for both starting and ending the enclosement
+            char endSymbol = optionalEndSymbol ?? startSymbol;
+
+            List<string> sections = [];
+            string remainder = input;
+
+            // If we are not searching for strings, any instances of start or end symbols enclosed within string delimeters are to be ignored
+            Dictionary<int, int> stringDelimeterIndexes = [];
+            if (startSymbol != syntaxConf.Symbols.Key.StringDelimeter)
+            {
+                stringDelimeterIndexes =  GetEnclosingCharacterIndexes(input, syntaxConf.Symbols.Key.StringDelimeter);
+            }
+
+            // Loop through the string and extract sections until no more start symbols are found
+            int startIndex;
+            do
+            {
+                startIndex = FindSymbolIndex(remainder, startSymbol, stringDelimeterIndexes);
+                int endIndex = FindSymbolIndex(remainder, endSymbol, stringDelimeterIndexes, startIndex + 1);
+                if (endIndex == -1)
+                {
+                    break;
+                }
+
+                string section = remainder[(startIndex + 1)..endIndex];
+                sections.Add(section);
+                remainder = remainder.Remove(startIndex, endIndex - startIndex + 1);
+            }
+            while (startIndex != -1);
+
+            return new ExtractSectionResult([.. sections], remainder);
+        }
+
+        /// <summary>
+        /// Extracts specifier parts from a parameter string
+        /// </summary>
+        /// <param name="input">The input string to extract from</param>
+        /// <param name="stringDelimeter">The delimeter that encloses the specifier part</param>
+        /// <param name="syntaxConf">Object that contains the symbols and tokens for structuring the file syntax. The syntax configuration is represented by a <see cref="PxFileSyntaxConf"/> object.</param>
+        /// <return>Returns an <see cref="ExtractSectionResult"/> object that contains the sections and the string that remains after the operation</return>
+        public static string[] GetSpecifiersFromParameter(string? input, char stringDelimeter, PxFileSyntaxConf syntaxConf)
```

Comment:
This method can be replaced with a line: xyz is null ? [] : ExtractSectionFromString(input, stringDelimeter, syntaxConf).Sections;